server.js

```diff
@@ -33,8 +33,10 @@ app.get('/search', (request, response) => {
 
 app.post('/show', searchHandler);
 app.post('/books', addBook);
-
-app.get('/details/:id', getOneBook);
+app.get('/details/:books_id', getOneBook);
+app.delete('/books/:books_id', deleteOneBook);
+app.put('/books/:books_id/edit', updateOneBook);
+app.get('/details/:books_id/edit', editOneBook);
```

Comment:
Personal preference would be to replace `details` in these routes with `books`. From a REST standpoint you want to be able to GET, PUT and DELETE the same resource/path (e.g. `/books/:book_id`).